truss/main.go

```diff
@@ -6,133 +6,209 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/pkg/errors"
 
-	"github.com/TuneLab/go-truss/truss/protostage"
+	"github.com/TuneLab/go-truss/truss/protostuff"
 	"github.com/TuneLab/go-truss/truss/truss"
 
 	"github.com/TuneLab/go-truss/deftree"
 	"github.com/TuneLab/go-truss/gendoc"
 	"github.com/TuneLab/go-truss/gengokit"
 )
 
-func main() {
+var (
+	pbPathFlag = flag.String("pbout", "", "The go package path where the protoc-gen-go .pb.go structs will be written.")
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]... [*.proto]...\n", os.Args[0])
+		flag.PrintDefaults()
+	}
+
 	flag.Parse()
 
 	if len(flag.Args()) == 0 {
-		exitIfError(errors.New("no arguments passed"))
+		fmt.Fprintf(os.Stderr, "%s: missing .proto file(s)\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Try '%s --help' for more information.\n", os.Args[0])
+		os.Exit(1)
 	}
+}
+
+func main() {
+	cfg, err := parseInput()
+	exitIfError(errors.Wrap(err, "could not parse input"))
+
+	dt, err := parseServiceDefinition(cfg.DefPaths)
+	exitIfError(errors.Wrap(err, "unable to parse input definition proto files"))
+
+	err = updateConfigWithService(cfg, dt)
+	exitIfError(errors.Wrap(err, "unable to read system based on service definition"))
```

Comment:
Haha, I probably should have looked more closely. But in this case, if you're going to wrap it, it should add value, and given the current error message, I don't think it does. "unable to read previously generated files" is clear enough, but if I got that message while running, there's a good chance I don't know why that's important, or what I need to do to fix it.

Looking at this example specifically, though, the error is only going to be returned if the next deeper level in the stack has a problem walking the directory tree. That's probably not going to happen, and if it does, I have three levels of errors without even wrapping it again:

1. unable to read previously generated files
2. could not to fully walk directory %s
3. error from `filepath.Walk`

I'd argue that this likely presents enough of a picture here without having to wrap again. We're not *obligated* to wrap errors in every level.

As an aside, we should keep in mind the language we're using in UI messages and be consistent. E.g., "could not" vs "unable to." Minor, but consistency in the user interface is a good thing.